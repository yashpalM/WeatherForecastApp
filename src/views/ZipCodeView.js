import React from 'react';
import { useState } from 'react';
import { View, Text, Button, Alert, StyleSheet, TouchableOpacity } from 'react-native';
import { Input } from 'react-native-elements';
import colors from '../constants/colors';

/**
 * ZipCodeView
 */

const ZipCodeView = props => {
    const [zipCode, setZipCode] = useState(``);
    const validation = (zipcode) => {
        if(zipcode.length <= 0){
            Alert.alert('ZIPCODE IS REQUIRED')
            return false;
        } else if (zipcode.length > 0 && zipcode.length < 6) {
            Alert.alert('ENTER A VALID ZIPCODE')
            return false;
        }
        return true;
    }
    return (
        <View style={styles.body}>
            <Input
                keyboardType={'numeric'}
                placeholder='Enter Zipcode'
                errorStyle={{ color: 'red' }}
                onChangeText={(zipcode) => setZipCode(zipcode)}
                onSubmitEditing={() => {
                    validation(zipCode) ? props.navigation.navigate('WeatherView', {
                        zipcode: zipCode
                    }) : ''
                }}
            />

            <TouchableOpacity style={styles.submitButton}
                onPress={() => {
                    validation(zipCode) ? props.navigation.navigate('WeatherView', { zipcode: zipCode })
                        : ''
                }}>
                <Text style={styles.submitText}> Submit </Text>
            </TouchableOpacity>

        </View>
    );
};

const styles = StyleSheet.create({
    body: {
        backgroundColor: '#C5C5C5',
        flex: 1,
        justifyContent: "center",
        alignContent: 'center',
        padding: 20
    },
    container: {
        marginTop: 32,
        paddingHorizontal: 24,
    },
    submitButton: {
        backgroundColor: colors.primary,
        padding: 10,
        margin: 15,
        height: 40,
    },
    submitText: {
        textAlign: 'center',
        color: colors.white,
        fontSize: 16,

    }
});

export default ZipCodeView;