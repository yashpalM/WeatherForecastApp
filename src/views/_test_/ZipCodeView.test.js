import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ZipCodeView from '../ZipCodeView';
import { Touchable, TouchableOpacity } from 'react-native';
import { Input } from 'react-native-elements';

jest.useFakeTimers()

let wrapper;
let props;

const createTestProps = (props) => ({
    navigation: {
        navigate: jest.fn()
    },
    ...props
});

beforeEach(() => {
    props = createTestProps({});
    wrapper = shallow(<ZipCodeView {...props} />);
});

it('Pincode screen renders correctly', () => {
    renderer.create(<ZipCodeView />).toJSON();
});

describe('ZipCodeView', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            expect(wrapper).toMatchSnapshot()
        });
    });
});

describe('Zincode validation', () => {
    function validation(zipcode) {
        if (zipcode.length <= 0) {
            return false;
        }
        else if (zipcode.length > 0 && zipcode.length < 6) {
            return false;
        }
        return true;
    }

    it('Pincode Empty Check', () => {
        expect(validation('')).toBe(false);
    })

    it('Pincode Length Check', () => {
        expect(validation('123')).toBe(false);
    })

    it('Pincode correct Check', () => {
        expect(validation('110030')).toBe(true);
    });
});

it('should render correctly', () => {
    expect(wrapper.find('View').length).toBe(1);
});

it('should render correctly', () => {
    expect(wrapper.find('Text').length).toBe(1);
});

it('should render correctly', () => {
    expect(wrapper.find(TouchableOpacity).length).toBe(1);
});

it('should render correctly', () => {
    expect(wrapper.find(Input).length).toBe(1);
});

it('onChangeText function check', () => {
    const onChangeText = jest.fn();

    wrapper.find(Input).first().props().onChangeText();

    // Check that our handler have been called 1 time
    expect(onChangeText).toHaveBeenCalledTimes(0);
});

it('onSubmitEditing function check', () => {
    const onSubmitEditing = jest.fn();

    wrapper.find(Input).first().props().onSubmitEditing(jest.fn());

    // Check that our handler have been called 1 time
    expect(onSubmitEditing).toHaveBeenCalledTimes(0);
});

it('onPress function check', () => {
    const onpress = jest.fn();

    wrapper.find(TouchableOpacity).first().props().onPress(jest.fn());

    // Check that our handler have been called 1 time
    expect(onpress).toHaveBeenCalledTimes(0);
});



