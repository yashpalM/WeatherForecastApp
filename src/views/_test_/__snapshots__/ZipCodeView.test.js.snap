// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZipCodeView Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
